### ISSUE TITLE
[Triage] Playlist index field not getting read for playlists

### ISSUE THREAD

**Issue by ckambler** [kieraneglin/pinchflat#409]
**Describe the bug**
I have a few playlists I would like to download, where they are university lectures arranged in chronological order. Sometimes the individual videos are not uploaded in order, so the playlist order is what matters for generating the correct file names to have Plex treat it like a season of a tv show "s01e01".

I tried using the {{ playlist_index }} field, but that either returned a value of 0 or NA. I saw on a previous issue #387  that the {{ media_playlist_index }} field is a workaround for this, but this also does not work for me. It names all files as being episode 0.

My Media profile output path template is:
{{ source_custom_name }}/Season 01/s01e{{ media_playlist_index }} - {{ title }}.{{ ext }}

as an example, the single video playllist here: https://www.youtube.com/playlist?list=PLUl4u3cNGP61O2kne_fe_Clsqvu0b3kcu
It should presumably label the only video there as episode 1, but for me it's always episode 0. I've tried this on other playlists, with the same result.

**Expected behavior**
I would expect that the {{ media_playlist_index }} field in my output format would sequentially number the videos by their order in the playlist.

**Diagnostic info**

- App Version: 2024.10.2
- yt-dlp Version: 2024.09.27
- Apprise Version: v1.9.0
- System Architecture: x86_64-pc-linux-gnu
- Timezone: America/Los_Angeles

**Comment by kieraneglin** [comment]
Hey there! Thanks for the report (:

I'll look into this! But just a heads up that if I don't figure it out tomorrow (Friday), it'll be at least a week until I get back to it since I'll be mostly out of cell service until next Sunday.

In any case, I'll let you know what I find!

**Comment by kieraneglin** [comment]
I've pushed an update that bumps `yt-dlp` - would you mind updating and letting me know if this issue is resolved?

**Comment by ckambler** [comment]
No, still looks like it isn't working. I updated to 2024.10.25, and just tried it, still showing s01e00 for the filename.

**Comment by kieraneglin** [comment]
I think I finally got it with #442! Just a heads up that, due to technical limitations, the source you're using `media_playlist_index` with will need to be deleted and re-added after this change

**Comment by ckambler** [comment]
just updated to the 2024.10.30 release, everything works as expected now. Thanks!

**Comment by nebula-it** [comment]
Hey @ckambler 
How do you use `playlist_index` in output template?
I've tried `/playlists//{{ source_custom_name }}/{{ playlist_index }}-{{ title }}.{{ ext }}` and `/playlists//{{ source_custom_name }}/%(playlist_index)s-{{ title }}.{{ ext }}` but they both result in final file nale starting with `NA`. Any idea?

**Comment by ckambler** [comment]
> Hey @ckambler How do you use `playlist_index` in output template? I've tried `/playlists//{{ source_custom_name }}/{{ playlist_index }}-{{ title }}.{{ ext }}` and `/playlists//{{ source_custom_name }}/%(playlist_index)s-{{ title }}.{{ ext }}` but they both result in final file nale starting with `NA`. Any idea?

I read somewhere in the documentation for pinchflat that {{ playlist_index }} won't work as intended because of how pinchflat fetches the videos from a playlist URL, so the workaround for that is the field {{ media_playlist_index }}. I use that and after the update discussed earlier in this thread, it works for me. My full output path template is below:

{{ source_custom_name }}/Season 01/s01e{{ media_playlist_index }} - {{ title }}.{{ ext }}

**Comment by nebula-it** [comment]
Thank you, that works
